experimental/packages/opentelemetry-instrumentation-http/test/utils/assertSpan.ts

```diff
@@ -96,49 +114,99 @@ export const assertSpan = (
+
+      if (
+        validations.resHeaders['content-encoding'] &&
+        validations.resHeaders['content-encoding'] !== 'identity'
+      ) {
+        assert.strictEqual(
+          span.attributes[ATTR_HTTP_RESPONSE_CONTENT_LENGTH],
+          contentLength
+        );
+      } else {
+        assert.strictEqual(
+          span.attributes[ATTR_HTTP_RESPONSE_CONTENT_LENGTH_UNCOMPRESSED],
+          contentLength
+        );
+      }
+    }
     assert.strictEqual(
-      span.attributes[ATTR_SERVER_ADDRESS],
+      span.attributes[ATTR_NET_PEER_NAME],
       validations.hostname,
       'must be consistent (PEER_NAME and hostname)'
     );
     if (!validations.noNetPeer) {
-      assert.ok(
-        span.attributes[ATTR_NETWORK_PEER_ADDRESS],
-        'must have PEER_IP'
-      );
-      assert.ok(span.attributes[ATTR_SERVER_PORT], 'must have PEER_PORT');
+      assert.ok(span.attributes[ATTR_NET_PEER_IP], 'must have PEER_IP');
+      assert.ok(span.attributes[ATTR_NET_PEER_PORT], 'must have PEER_PORT');
     }
     assert.ok(
-      (span.attributes[ATTR_URL_FULL] as string).indexOf(
-        span.attributes[ATTR_SERVER_ADDRESS] as string
+      (span.attributes[ATTR_HTTP_URL] as string).indexOf(
+        span.attributes[ATTR_NET_PEER_NAME] as string
       ) > -1,
       'must be consistent'
     );
   }
   if (span.kind === SpanKind.SERVER) {
+    if (validations.reqHeaders && validations.reqHeaders['content-length']) {
+      const contentLength = validations.reqHeaders['content-length'];
+
+      if (
+        validations.reqHeaders['content-encoding'] &&
+        validations.reqHeaders['content-encoding'] !== 'identity'
+      ) {
+        assert.strictEqual(
+          span.attributes[ATTR_HTTP_REQUEST_CONTENT_LENGTH],
+          contentLength
+        );
+      } else {
+        assert.strictEqual(
+          span.attributes[ATTR_HTTP_REQUEST_CONTENT_LENGTH_UNCOMPRESSED],
+          contentLength
+        );
+      }
+    }
+    if (validations.serverName) {
+      assert.strictEqual(
+        span.attributes[ATTR_HTTP_SERVER_NAME],
+        validations.serverName,
+        ' must have serverName attribute'
+      );
+      assert.ok(span.attributes[ATTR_NET_HOST_PORT], 'must have HOST_PORT');
+      assert.ok(span.attributes[ATTR_NET_HOST_IP], 'must have HOST_IP');
+    }
     assert.strictEqual(
-      span.attributes[ATTR_URL_SCHEME],
+      span.attributes[ATTR_HTTP_SCHEME],
       validations.component,
       ' must have http.scheme attribute'
     );
     assert.ok(typeof span.parentSpanContext?.spanId === 'string');
     assert.ok(isValidSpanId(span.parentSpanContext.spanId));
 
-    assert.strictEqual(
-      span.attributes[ATTR_URL_PATH],
-      validations.path || validations.pathname
-    );
+    // TODO: stable only
+    // assert.strictEqual(
+    //   span.attributes[ATTR_URL_PATH],
+    //   validations.path || validations.pathname
+    // );
```

Comment:
ATTR_URL_PATH presence is being tested by `describe('with semconv stability set to http'` in http-enable.test.ts, so I'm fine removing this block.